src/main/java/beans/menus/EntityListResponse.java

```
@@ -0,0 +1,133 @@
+    private Entity[] entities;
+    private DisplayElement action;
+    private Style[] styles;
+
+    public EntityListResponse(){}
+
+    public EntityListResponse(EntityScreen nextScreen) {
+        SessionWrapper session = nextScreen.getSession();
+        Detail shortDetail = nextScreen.getShortDetail();
+        EvaluationContext ec = session.getEvaluationContext();
+        Vector<TreeReference> references = ec.expandReference(((EntityDatum)session.getNeededDatum()).getNodeset());
+        processTitle(session);
+        processEntities(nextScreen, references, ec);
+        processStyles(shortDetail);
+        processActions(shortDetail, nextScreen.getSession());
+    }
+
+    private void processTitle(SessionWrapper session) {
+        setTitle(SessionUtils.getBestTitle(session));
+    }
+
+    private void processEntities(EntityScreen screen, Vector<TreeReference> references, EvaluationContext ec) {
+        entities = new Entity[references.size()];
+        int i = 0;
+        for (TreeReference entity : references) {
+            Entity newEntity = processEntity(entity, screen, ec);
+            newEntity.setDetail(new EntityDetailResponse(processDetails(screen, ec, entity)));
+            entities[i] = newEntity;
+            i++;
+        }
+    }
+
+    private Entity processEntity(TreeReference entity, EntityScreen screen, EvaluationContext ec) {
+        Detail detail = screen.getShortDetail();
+        Entity ret = new Entity();
+        EvaluationContext context = new EvaluationContext(ec, entity);
+
+        detail.populateEvaluationContextVariables(context);
+
+        DetailField[] fields = detail.getFields();
+        Object[] data = new Object[fields.length];
+
+        int i = 0;
+        for (DetailField field : fields) {
+            Object o;
+            try {
+                o = field.getTemplate().evaluate(context);
+            } catch(XPathException e) {
+                o = "error (see output)";
```

Yeah I think we should actually just fail hard here